integration-tests/src/tests/client/features/yield_timeouts.rs

```diff
@@ -0,0 +1,366 @@
+
+    // By introducing congestion, we can delay the yield timeout
+    for block_height in NEXT_BLOCK_HEIGHT_AFTER_SETUP..(YIELD_TIMEOUT_HEIGHT + 3) {
+        // Submit txns to congest the block at height YIELD_TIMEOUT_HEIGHT and delay the timeout
+        if block_height == YIELD_TIMEOUT_HEIGHT - 1 {
+            create_congestion(&mut env);
+        }
+
+        env.produce_block(0, block_height);
+
+        // The transaction will not have a result until the timeout is reached
+        assert_eq!(
+            env.clients[0].chain.get_partial_transaction_result(&yield_tx_hash).unwrap().status,
+            FinalExecutionStatus::Started
+        );
+    }
+
+    // Advance more blocks so that the congestion clears and the yield callback is executed.
+    for i in 0..10 {
+        env.produce_block(0, YIELD_TIMEOUT_HEIGHT + 3 + i);
+    }
+
+    assert_eq!(
+        env.clients[0].chain.get_partial_transaction_result(&yield_tx_hash).unwrap().status,
+        FinalExecutionStatus::SuccessValue(vec![0u8]),
+    );
+}
+
+/// In this case we invoke yield_resume at the last block possible.
+#[test]
+fn yield_resume_just_before_timeout() {
+    let yield_payload = vec![6u8; 16];
+    let (mut env, yield_tx_hash, data_id) = prepare_env_with_yield(yield_payload.clone(), None);
+    assert!(NEXT_BLOCK_HEIGHT_AFTER_SETUP < YIELD_TIMEOUT_HEIGHT);
+
+    for block_height in NEXT_BLOCK_HEIGHT_AFTER_SETUP..YIELD_TIMEOUT_HEIGHT {
+        // Submit txn so that yield_resume is invoked in the block at height YIELD_TIMEOUT_HEIGHT
+        if block_height == YIELD_TIMEOUT_HEIGHT - 1 {
+            invoke_yield_resume(&mut env, data_id, yield_payload.clone());
+        }
+
+        env.produce_block(0, block_height);
+
+        // The transaction will not have a result until the yield execution is resumed
+        assert_eq!(
+            env.clients[0].chain.get_partial_transaction_result(&yield_tx_hash).unwrap().status,
+            FinalExecutionStatus::Started
+        );
+    }
+
+    // In this block the `yield_resume` host function is invoked, producing a YieldResume receipt.
+    env.produce_block(0, YIELD_TIMEOUT_HEIGHT);
+    assert_eq!(
+        env.clients[0].chain.get_partial_transaction_result(&yield_tx_hash).unwrap().status,
+        FinalExecutionStatus::Started
+    );
+    // Here we expect two receipts to be produced; one from yield_resume and one from timeout.
+    assert_eq!(find_yield_data_ids_from_latest_block(&env), vec![data_id, data_id]);
+
+    // In this block the resume receipt is applied and the callback is executed with the resume payload.
+    env.produce_block(0, YIELD_TIMEOUT_HEIGHT + 1);
+    assert_eq!(
+        env.clients[0].chain.get_partial_transaction_result(&yield_tx_hash).unwrap().status,
+        FinalExecutionStatus::SuccessValue(vec![16u8]),
+    );
+}
+
+/// In this test we introduce congestion to delay the yield timeout so that we can invoke
+/// yield resume after the timeout height has passed.
```

Comment:
We look for the invocation of the continuation function here, and not a transaction with the yield_resume host function after the timeout has occurred, right?

Reply:
Yes, we verify that both things occur. The continuation function verifies that it receives the right payload (and not an empty one, indicating invocation by timeout).